components/stepped-form/index.tsx

```diff
@@ -159,189 +159,239 @@ const Icon = styled(FontAwesomeIcon)`
-        break;
-      case "EMAIL":
-        inputType = "email";
-        break;
-      case "TELEPHONE":
-        inputType = "tel";
-        break;
-      case "SECURE_INPUT":
-        inputType = "password";
+  const handleSubmit = async (props) => {
+    const event = props.event;
+    event.preventDefault();
+    // array of objects containing id and value from input
+    const responses = [];
+    // map out data from checkbox responses
+    const checkboxMap = {}
+    // map out data from ranking responses
+    const rankingMap = {"0": {"0": "0"}}
+    for (const response of event.target) {
+      if (response.value) {
+        switch (response.type) {
+          case "select-one":
+            // handle rank type
+            if (response.name.split("-").length === 3) {
+              const split = response.name.split("-")
+              const responseId = split[0]
+              const value = split[1]
+              const idx = `${+response.value - 1}`
+              if (!rankingMap[responseId]) {
+                rankingMap[responseId] = {}
+              }
+              rankingMap[responseId][idx] = value
+              break;
+            }
+            responses.push({id: +response.name, response: +response.value});
+            break;
+          case "dropdown":
+            responses.push({id: response.name, response: +response.value});
+            break;
+          case "checkbox":
+            const responseId = response.name.split("-")[0];
+            if (response.checked) {
+              checkboxMap[responseId] ? checkboxMap[responseId].push(+response.value) : checkboxMap[responseId] = [+response.value];
+            }
+            break;
+          case "date":
+            if (response.value) {
+              responses.push({id: +response.name, response: new Date(response.value)});
+            }
+            break;
+          case "submit":
+            break;
+          case "password":
+          case "text":
+          case "textarea":
+          case "email":
+          case "tel":
+          default:
+            responses.push({id: +response.name, response: response.value});
+            break;
+        }
+      }
+    }
+    // handle creating checkbox responses
+    for (const id in checkboxMap) {
+      responses.push({id: +id, response: checkboxMap[id]})
+    }
+    // handle creating ranking responses
+    for (const id in rankingMap) {
+      if (id === '0') {
+        continue
+      }
+      const rankingResponse = []
+      for (const index in rankingMap[id]) {
+        const value = rankingMap[id][index]
+        rankingResponse.splice(+index, 1, +value)
+      }
+      responses.push({id: +id, response: rankingResponse})
+    }
+    const request = new SubmitFormDto()
+    request.responses = responses;
+    // Submit!
+    await postFormSubmission('1', '1', {responses})
+    // #TODO - redirect home and try-catch with rendered error message "Something went wrong, try again (err: error_code)"
```

Comment:
In all my react native stuff it's usually
```
catch(error) {
  Alert.message('something went wrong, ${error.message}')
}
```
Or at least something along those linez